rdflib/namespace.py

```diff
@@ -330,6 +330,21 @@ def compute_qname(self, uri, generate=True):
             namespace, name = split_uri(uri)
             namespace = URIRef(namespace)
             prefix = self.store.prefix(namespace)
+            # check to see if there are longer prefixes defined
+            temp_namespace = namespace.toPython()
+            output_prefix = prefix
+            output_namespace = namespace
+            output_name = name
+            for i in xrange(1, len(name) + 1):
```

Comment:
Turns out string.split is very, very slow. New implementation using len and slicing gives 1.162s using the same method as above.